### PR TITLE
Added AudioMNIST dataset.

### PR DESCRIPTION
**Pull request by harritaylor**

### CODE REVIEW

File: torchaudio/datasets/AudioMNIST.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    def __init__(self, root, split, train=True, transform=None, target_transform=None, download=False):
+        self.root = os.path.expanduser(root)
+        self.transform = transform
+        self.target_transform = target_transform
+        self.train = train  # training set or test set
+
+        if download:
+            self.download()
+
+        # if not self._check_exists():
```
Comment (soumith) [inline torchaudio/datasets/AudioMNIST.py]: why is this commented?

Comment (harritaylor) [reply torchaudio/datasets/AudioMNIST.py]: Sorry, it shouldn't be commented out - I'm not sure why it is.

**Comment by cpuhrsch** [comment]
@harritaylor - would you still be willing to resolve the comments and then have us merge this code?

**Comment by harritaylor** [comment]
> @harritaylor - would you still be willing to resolve the comments and then have us merge this code?

Yes, sorry - I'll see if I have some time this weekend to sort it out.

**Comment by harritaylor** [comment]
@soumith @cpuhrsch Sorry about the long delay on this, I've been preoccupied with other work. I'd like to rewrite this code and resubmit a pull but I'm not sure the best way of doing that. Is it best to get rid of this pull request and submit a new one when ready?

**Comment by Mistobaan** [comment]
@harritaylor you can just close the PR at the bottom of the thread, and submit a new one when is ready.

**Comment by vincentqb** [comment]
I'll close this PR for now, but please feel free to open a new one when ready :)

PS. I'll link pytorch/pytorch#24915 since we are discussing new dataset constructs.

**Comment by Sairam954** [comment]
Do you have any update on including AudioMNIST to torchaudio

**Pull request closed without merging**